GVFS/FastFetch/NativeMacMethods.cs

```diff
@@ -0,0 +1,51 @@
+﻿using GVFS.Common.Tracing;
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace FastFetch
+{
+    public class NativeMacMethods
+    {
+        public const int ReadOnly = 0x0000;
+        public const int WriteOnly = 0x0001;
+
+        public const int Create = 0x0200;
+        public const int Truncate = 0x0400;
+
+        public static unsafe void WriteFile(ITracer tracer, byte* originalData, long originalSize, string destination, ushort mode)
+        {
+            int fileDescriptor = -1;
+            try
+            {
+                fileDescriptor = Open(destination, WriteOnly | Create | Truncate, mode);
+                if (fileDescriptor != -1)
+                {
+                    IntPtr result = Write(fileDescriptor, originalData, (IntPtr)originalSize);
+                    if (result.ToInt32() == -1)
+                    {
+                        throw new Win32Exception(Marshal.GetLastWin32Error());
+                    }
+                }
+            }
+            catch (Win32Exception e)
+            {
+                tracer.RelatedError("Error writing file {0}. Win32Exception: {1}", destination, e);
+                throw;
+            }
+            finally
+            {
+                Close(fileDescriptor);
+            }
+        }
+
+        [DllImport("libc", EntryPoint = "open", SetLastError = true)]
+        public static extern int Open(string path, int flag, int creationMode);

```

Comment:
I think that `creationMode` should be a `ushort`.

From the `open` manpage:

> The oflag argument may indicate that the file is to be created if it does not exist (by specifying the O_CREAT flag).  In this case, open() and openat() require an additional argument mode_t mode; the file is created with mode mode as described in chmod(2) and modified by the process' umask value (see umask(2)).

*_mode_t.h*
`typedef	__darwin_mode_t		mode_t;`

*_types.h*
`typedef __uint16_t	__darwin_mode_t;	/* [???] Some file attributes */`